Dockerfile

```diff
@@ -0,0 +1,41 @@
+# The MIT License
+#
+#  Copyright 2020 Loghi Perinpanayagam.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to deal
+#  in the Software without restriction, including without limitation the rights
+#  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#  copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+
+
+FROM jenkins/ssh-agent
+
+# install python
+RUN apt-get update  && apt-get install -y \
+    python \
+    python-pip
+
+RUN apt install default-jdk -y
+
+# install dependencies
+RUN pip install ipython \
+    ipykernel \
+    jupyter-client \
+    protobuf \
+    grpcio
```

Comment:
While I was trying to build using these cmd `--no-install-recommends` and `rm -rf /var/lib/apt/lists/*`, installing grpcio is failing

Reply:
It probably needs some dev lib. You can workaround that by - I think - doing the following in a single `RUN` command:

- apt update
- apt install
- pip install
- remove apt files (not sure if there's any pip cache that can be removed)

This should give you a single layer, and smaller image size (you can build before, look at its size, and re-build with this to compare and see how many MB's you saved)